### ISSUE TITLE
SonarQube not ignoring C++ code with // NOSONAR

### ISSUE THREAD

**Issue by alexbde** [SonarOpenCommunity/sonar-cxx#885]
We're using `SonarQube 5.3`. One of the developers told SonarQube is _not_ ignoring a false positive although he marked it with `// NOSONAR` in source code.

[![SonarQube not ignoring NOSONAR][1]][1]

In Java it's working fine. Is there another way for C++? We're using `cppcheck` plugin.

I know the way marking it as false positive via UI. But it would be good to mark it right in the source code.

PS: This question was originally posted on StackOverflow but I got forwarded to this project: http://stackoverflow.com/questions/37788263/sonarqube-not-ignoring-c-code-with-nosonar

  [1]: http://i.stack.imgur.com/5BBBV.gif

**Comment by guwirth** [comment]
The static code analyser `cppcheck` is not supporting `// NOSONAR`.The C++ plugin is reading the resulting report and display the issues.

See cppcheck manual: To suppress the error message, a comment can be added:
```C++
void f() {
   char arr[5];
   // cppcheck-suppress arrayIndexOutOfBounds
   arr[10] = 0;
}
```
